packages/hoppscotch-app/helpers/import-export/export/openapi.ts

```diff
@@ -0,0 +1,406 @@
+            ...generateEntries(value as object),
+          },
+        }
+      : {
+          default: value,
+        },
+  ]
+}
+
+const generateEntries = (body: object): OpenAPIV3.SchemaObject["properties"] =>
+  pipe(
+    Object.entries(body),
+    A.map(convertHoppRestReqBodyEntryToOpenApiBodyEntry),
+    tupleToRecord
+  )
+
+const generateOpenApiHeaders = (
+  hoppHeaders: HoppRESTHeader[]
+): OpenAPIV3.ParameterObject[] =>
+  pipe(
+    hoppHeaders,
+    A.map(
+      (header): OpenAPIV3.ParameterObject => ({
+        in: "header",
+        name: header.key,
+        schema: {
+          default: header.value,
+        },
+      })
+    )
+  )
+
+const generateOpenApiQueryParams = (
+  hoppParams: HoppRESTParam[]
+): OpenAPIV3.ParameterObject[] =>
+  pipe(
+    hoppParams,
+    A.map((param) => ({
+      in: "query",
+      name: param.key,
+      schema: {
+        default: param.value,
+      },
+    }))
+  )
+
+const isValidContentType = (
+  contentType: unknown
+): contentType is keyof typeof knownContentTypes =>
+  !!(
+    contentType &&
+    typeof contentType === "string" &&
+    objHasProperty(contentType, "string")(knownContentTypes)
+  )
+
+const isNonFormDataBody = (
+  body: HoppRESTReqBody
+): body is HoppRESTReqBodyNonFormData =>
+  !!(
+    body.body &&
+    body.contentType &&
+    body.contentType !== "multipart/form-data"
+  )
+
+const isFormDataBody = (
+  body: HoppRESTReqBody
+): body is HoppRESTReqBodyFormData => body.contentType === "multipart/form-data"
+
+const isEmptyBody = (body: HoppRESTReqBody): body is HoppRestReqBodyEmpty =>
+  !(body.body && body.contentType)
+
+type RequestBodyGenerationError = "INVALID_CONTENT_TYPE" | "INVALID_BODY"
+
+/**
+ * generates the openapi document body for different content types
+ * 1. when the body and content type are null, we return an empty openapi body
+ * 2. when the content type is an invalid one, we return INVALID_CONTENT_TYPE error
+ * 3. when the content type is json based content types, we convert that to openapi properties recursively
+ * 4. multipart/formdata is also handled differently
+ * 5. in other valid content types, the body is handled as a string
+ */
+export const generateOpenApiRequestBody = (
+  hoppRequestBody: HoppRESTReqBody
+): E.Either<RequestBodyGenerationError, OpenAPIV3.RequestBodyObject> =>
+  pipe(
+    hoppRequestBody,
+    E.fromPredicate(isEmptyBody, () => "NON_EMPTY_BODY" as const),
+    E.map(
+      (): OpenAPIV3.RequestBodyObject => ({
+        content: {},
+      })
+    ),
+    E.altW(() =>
+      pipe(
+        hoppRequestBody.contentType,
+        E.fromPredicate(
+          isValidContentType,
+          () => "INVALID_CONTENT_TYPE" as const
+        ),
+        E.chainW((contentType) =>
+          pipe(
+            hoppRequestBody,
+            O.fromPredicate(isNonFormDataBody),
+            O.chain((hoppRequestBody) =>
+              pipe(
+                hoppRequestBody,
+                O.fromPredicate(
```

Comment:
`pipe` can be replaced by a `flow`.

```suggestion
            O.chain(
              flow(
                O.fromPredicate(
```